File: engine/model/src/main/scala/tools/packet/monster/CancelMonsterStatus.scala
package tools.packet.monster

import client.status.MonsterStatus
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class CancelMonsterStatus(private var _objectId: Int, private var _stats: java.util.Map[MonsterStatus, java.lang.Integer]) extends PacketInput {
  def objectId: Int = _objectId

  def stats: java.util.Map[MonsterStatus, java.lang.Integer] = _stats

  override def opcode(): SendOpcode = SendOpcode.CANCEL_MONSTER_STATUS
}

File: engine/engine-model/src/main/java/tools/packet/monster/CancelMonsterStatus.java
package tools.packet.monster; 
import java.util.Map;

import client.status.MonsterStatus;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record CancelMonsterStatus( Integer objectId,  Map<MonsterStatus, Integer> stats) implements PacketInput {
  @Override
  public SendOpcode opcode() {
    return SendOpcode.CANCEL_MONSTER_STATUS;
  }
}